src/tribol/interface/tribol.cpp

```diff
@@ -921,6 +1110,96 @@ integer update( integer cycle, real t, real &dt )
       }
 
       CouplingScheme* couplingScheme  = csManager.getCoupling(csIndex);
+
+      // update redecomp meshes if supplied mfem data
+      if (couplingScheme->hasMfemData())
+      {
+         auto mfem_data = couplingScheme->getMfemMeshData();
+         axom::Array<int> mesh_ids {2, 2};
+         mesh_ids[0] = mfem_data->GetMesh1ID();
+         mesh_ids[1] = mfem_data->GetMesh2ID();
+         mfem_data->UpdateMeshData();
+         auto coord_ptrs = mfem_data->GetRedecompCoordsPtrs();
+
+         // save penalty data for old meshes
+         auto& mesh_manager = MeshManager::getInstance();
+         axom::Array<MeshElemData> mesh_data {0, 2};
+         mesh_data.push_back(mesh_manager.GetMeshInstance(mesh_ids[0]).m_elemData);
+         mesh_data.push_back(mesh_manager.GetMeshInstance(mesh_ids[1]).m_elemData);
+
+         registerMesh(
+            mesh_ids[0],
+            mfem_data->GetMesh1NE(),
+            mfem_data->GetNV(),
+            mfem_data->GetMesh1Conn(),
+            mfem_data->GetElemType(),
+            coord_ptrs[0],
+            coord_ptrs[1],
+            coord_ptrs[2]
+         );
+         registerMesh(
+            mesh_ids[1],
+            mfem_data->GetMesh2NE(),
+            mfem_data->GetNV(),
+            mfem_data->GetMesh2Conn(),
+            mfem_data->GetElemType(),
+            coord_ptrs[0],
+            coord_ptrs[1],
+            coord_ptrs[2]
+         );
+
+         // apply penalty data
+         for (int i {0}; i < 2; ++i)
+         {
+            auto& new_mesh_data = mesh_manager.GetMeshInstance(mesh_ids[i]).m_elemData;
```

Comment:
Is all of this needed? I'm still confused about getting penalty data from an API call vs what looks like grabbing it from prior mesh data? What if it changes? Also this looks circular? new_mesh_data points to m_elemData on mesh_manager with mesh_ids[i], but mesh_data[i] is the m_elemData off the same mesh_manager with mesh_ids[i]. Maybe I'm missing something. Nonetheless, seeing a copying of penalty data in tribol::update() is confusing to me.

Reply:
Removed